test/Microsoft.AspNetCore.Razor.Design.Test/IntegrationTests/ProjectDirectory.cs

```diff
@@ -108,9 +116,45 @@ private ProjectDirectory(string directoryPath)
 
         public string DirectoryPath { get; }
 
+        public string GetFullPath(params string[] pathSegments)
+        {
+            var path = Path.Combine(pathSegments);
+            path = Path.GetFullPath(Path.Combine(DirectoryPath, path));
+            return path;
+        }
+
+        public IDisposable LockForReadWrite(string directory, string pattern)
+        {
+            directory = Path.GetFullPath(Path.Combine(DirectoryPath, directory));
+            var disposables = new List<IDisposable>();
+            foreach (var file in Directory.EnumerateFiles(directory, pattern, SearchOption.AllDirectories))
+            {
+                disposables.Add(LockForReadWrite(file));
+            }
+
+            var disposable = new Mock<IDisposable>();
+            disposable.Setup(d => d.Dispose())
+                .Callback(() => disposables.ForEach(d => d.Dispose()));
+
+            return disposable.Object;
+        }
```

Comment:
No, we shouldn't have/need this. Each test should operate on its own copy and be independeny

Reply:
Right. This is meant to ensure incremental builds (see https://github.com/aspnet/Razor/pull/1857/files#diff-7ceaebc6353fabd862893aa68747108bR25)

Reply:
Yeah I gets it now. This should be cleaned up, but the idea is good.